code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm

```diff
@@ -102,9 +102,10 @@ Doesn't work on other aliens/AI.*/
 	action_icon_state = "alien_whisper"
 
 /obj/effect/proc_holder/alien/whisper/fire(mob/living/carbon/user)
-	var/mob/living/M = input("Select who to whisper to:","Whisper to?",null) as mob in oview(user)
-	if(!M)

```

Comment:
Ignoring all your other changes, you could've just changed this line to if (!istype(M)). It will then check if it is null and check if it is the type that M is set to be (mob/living in this case).